### PR TITLE
Update GCSAdapter methods to reflect latest client API

### PR DESCRIPTION
**Pull request by pmmlo**

### CODE REVIEW

**Comment by pmmlo** [comment]
@TomWFox Here's the PR in a commit, if you could take a look.  Thanks

**Comment by TomWFox** [comment]
Thanks for the PR! Unfortunately I'm not best placed to review this.

**Comment by motte** [comment]
> Thanks for the PR! Unfortunately I'm not best placed to review this.

No worries!  Will try someone else.

**Comment by pmmlo** [comment]
> Thanks for the PR! Unfortunately I'm not best placed to review this.

No worries!  Thanks for sending over to core-mainters.  Cheers!

File: index.js [hunk 1]
```diff
@@ -55,71 +56,49 @@ function GCSAdapter() {
...
+	params,
     });
-    uploadStream.write(data);
-    uploadStream.end();
-  });
+
+    console.log(`${filename} uploaded to ${bucketName}.`);
+  }
+
+  uploadFile().catch(console.error);
```
Comment (davimacedo) [inline index.js:78]: I believe we should await. And also not only output the error to the console but let it be thrown.

File: index.js [hunk 2]
```diff
@@ -55,71 +56,49 @@ function GCSAdapter() {
...
-      }
-      resolve(res);
-    });
-  });
+    async function deleteFile() {
+      await this._gcsClient.bucket(this._bucket).file(this._bucketPrefix + filename).delete();
+      console.log(`${this._bucket} ${filename} deleted.`);
+    }
+
+    deleteFile().catch(console.error);
```
Comment (davimacedo) [inline index.js:87]: same thing here

File: index.js [hunk 3]
```diff
@@ -55,71 +56,49 @@ function GCSAdapter() {
...
-  });
+    async function downloadFile() {
+      const options = {
+        destination: filename,
+      };
+      await this._gcsClient.bucket(this._bucket).file(this._bucketPrefix + filename).download(options);
+
+      console.log(`${this._bucket} ${filename} downloaded.`);
+    }
+    downloadFile().catch(console.error);
```
Comment (davimacedo) [inline index.js:101]: here we also need to return the data

**Review by davimacedo — CHANGES_REQUESTED** [review]
Please find  some comments below. Would you mind to share a link with the changes of the last API? Also, could you please run tests and fix the errors?

**Comment by pmmlo** [comment]
> Please find some comments below. Would you mind to share a link with the changes of the last API? Also, could you please run tests and fix the errors?

@davimacedo Thanks for taking a look.

I agree about the await and returning on file upload.  I just wanted to keep it as close to original as possible for the first PR.  I think the original file upload also did not return anything.

I actually am not setup to use this adapter and fully test it.  I'll setup a demo account and reopen when I have more time.

Cheers!

**Pull request closed without merging**